web/services/clusters_test.go

```diff
@@ -96,7 +96,7 @@ func (suite *ClustersServiceTestSuite) TestClustersService_GetAll() {
 	suite.checksService.On("GetAggregatedChecksResultByCluster", "2").Return(&AggregatedCheckData{WarningCount: 1}, nil)
 	suite.checksService.On("GetAggregatedChecksResultByCluster", "3").Return(&AggregatedCheckData{CriticalCount: 1}, nil)
 
-	clusters, _ := suite.clustersService.GetAll(map[string][]string{})
+	clusters, _ := suite.clustersService.GetAll(nil, nil)
```

Comment:
I find it a bit meh having to call `.GetAll(nil. nil)` to get the complete list.
I'd maybe design differently the interface with a method that actually gets all the clusters, and one for filtered purposes.

Yet I understand why you did this, so it's a good iteration.

Reply:
I have to say i did a bit of reasearch  and the use of nil is pretty common also in the golang standard library.
There are other tricks with variadic arugments but they mean a lot of boiler code that I'm not confident will be helpful at the end.
On the other hand, adding another method will add more code to test and mantain, so again the idea is to minimize that since we are going to use that method with filtering and pagination 99.9% of the time.

Reply:
@nelsonkopliku I think the same about it, but bear in mind that ignoring the "nil is evil" line of thoughs is a cornerstone in Golang's design. Hence, the researches done by @fabriziosestito show something that I absolutely dislike, but extremely idiomatic at the same time :smile:

Reply:
@nelsonkopliku @dottorblaster see the edit of my comment, i was also biased against nil eheh if we find a better solution i am all hear but i could not!

Reply:
makes sense guys, thanks for clarification. 
Sometimes I still have to wrap my head around some go quirks, but that's ok, just need to deal with it :sweat_smile: